src/services/userService.js

```diff
@@ -0,0 +1,45 @@
+let users = [];
+
+const resetUsers = () => {
+  users = [];
+};
+
+const getAllUsers = () => {
+  return users;
+};
+
+const getUserById = (id) => {
+  return users.find((person) => person.id === +id) || null;
+};
+
+const createUser = (name) => {
+  const user = {
+    id: Math.floor(Math.random() * 1000),
+    name,
+  };
+
+  users.push(user);
+
+  return user;
+};
+
+const updateUser = ({ id, name }) => {
+  const user = getUserById(+id);
+
+  Object.assign(user, { name });
+
+  return user;
```

Comment:
Here you're directly modifying the user object. What if the user with the provided `id` doesn't exist? It would be better to first check if the user exists. If not, throw an error. Only after that, update the user.